src/commands/switch.js

```diff
@@ -0,0 +1,99 @@
+const { flags } = require('@oclif/command');
+const inquirer = require('inquirer');
+const AbstractAuthenticatedCommand = require('../abstract-authenticated-command');
+const BranchManager = require('../services/branch-manager');
+const ProjectManager = require('../services/project-manager');
+const withCurrentProject = require('../services/with-current-project');
+const envConfig = require('../config');
+
+class SwitchCommand extends AbstractAuthenticatedCommand {
+  async selectBranch(branches) {
+    try {
+      const { branch } = await inquirer.prompt([{
+        name: 'branch',
+        message: 'Select the branch you want to set current',
+        type: 'list',
+        choices: [
+          // NOTICE: Current branch should be last dispalyed branch.
+          ...branches.filter((currentBranch) => !currentBranch.isCurrent),
+          ...branches.filter((currentBranch) => currentBranch.isCurrent),
+        ].map((currentBranch) => currentBranch.name),
+      }]);
+
+      return branch;
+    } catch (error) {
+      const customError = BranchManager.handleBranchError(error);
+
+      return this.error(customError);
+    }
+  }
+
+  async switchTo(selectedBranch, environmentSecret) {
+    try {
+      await BranchManager.switchBranch(selectedBranch, environmentSecret);
+
+      return this.log(`✅ Switched to branch: ${selectedBranch.name}.`);
+    } catch (error) {
+      const customError = BranchManager.handleBranchError(error);
+
+      return this.error(customError);
+    }
+  }
+
+  async getConfig(envSecret) {
+    const parsed = this.parse(SwitchCommand);
+    const commandOptions = { ...parsed.flags, ...parsed.args, envSecret };
+
+    const config = await withCurrentProject({ ...envConfig, ...commandOptions });
+
+    if (!config.envSecret) {
+      const environment = await new ProjectManager(config)
+        .getDevelopmentEnvironmentForUser(config.projectId);
+      config.envSecret = environment.secretKey;
+    }
+
+    return config;
+  }
+
+  async runIfAuthenticated() {
+    try {
+      const envSecret = process.env.FOREST_ENV_SECRET;
+      const config = await this.getConfig(envSecret);
+      const branches = await BranchManager.getBranches(envSecret) || [];
```

Comment:
Here the command crash when i use it outside a project.
If you get the envSecret from the config it should be work.
```suggestion
      const branches = await BranchManager.getBranches(config.envSecret) || [];
```